index.js

```diff
@@ -3,127 +3,80 @@
 var mongoose = require('mongoose');
 var bson = require('bson');
 var util = require('util');
-var uuid = require('node-uuid');
+var uuidParse = require('uuid-parse');
 
 var Document = mongoose.Document;
 
-function getter (binary){
-  if(!binary) return undefined;
-  if(!(binary instanceof mongoose.Types.Buffer.Binary)) return binary;
+function getter(binary) {
+  if (binary == null) return undefined;
+  if (!(binary instanceof mongoose.Types.Buffer.Binary)) return binary;
 
-  
   var len = binary.length();
-
   var b = binary.read(0,len);
-
   var buf = new Buffer(len);
+  var hex = '';
```

Comment:
For L17-33 ,  Why not use https://www.npmjs.com/package/uuid-parse#uuidparseunparsebuffer-offset    ? 

```javascript
function getter(binary) {
  const string = uuidParse.unparse(bytes); // -> '797ff043-11eb-11e1-80d6-510998755d10'
```

or failing that:  https://nodejs.org/api/buffer.html#buffer_class_method_buffer_from_string_encoding
```javascript
const buf2 = Buffer.from('7468697320697320612074c3a97374', 'hex');
```

Reply:
Originally, I was just trying to do the minimum amount of work to get the dependencies updated. I hadn't really scrutinized the implementation. Though, now that we are those are certainly great alternatives. I'll update with a cleaner implementation.